### PR TITLE
Update installation steps for postgresql

### PR DESCRIPTION
**Pull request by SergioAtSUSE**

### CODE REVIEW

File: docs/Installing.asciidoc [hunk 1]
```diff
@@ -99,10 +99,50 @@ You can do that in +/etc/openqa/openqa.ini+:
...
+
+To setup openQA you will need a PostgreSQL database available to the user
+running openQA (geekotest).
+
+You need to create a database (openqa) and a dedicated user account (geekotest)
+with full access to it. To configure access to the chosen database in openQA,
+edit +/etc/openqa/database.ini+ and change the settings in the +[production]+
+section.
+
+[source.ini]
```
Comment (okurz) [inline docs/Installing.asciidoc]: I guess that should be `[source,sh]` instead. The following is not a file content but commands to execute

File: docs/Installing.asciidoc [hunk 2]
```diff
@@ -120,6 +160,8 @@ openQA runs on each boot, you should also +systemctl enable+ the same services.
 
 [source,sh]
 --------------------------------------------------------------------------------
+systemctl enable postgresql
+systemctl enable openqa-resource-allocator
```
Comment (coolo) [inline docs/Installing.asciidoc]: the openqa-webui.service requires  openqa-resource-allocator.service openqa-scheduler.service openqa-websockets.service - as such they don't need manual starting.

File: docs/Installing.asciidoc [hunk 3]
```diff
@@ -99,10 +99,50 @@ You can do that in +/etc/openqa/openqa.ini+:
 httpsonly = 0
 --------------------------------------------------------------------------------
 
+=== Database engine (PostgreSQL)
+
+To setup openQA you will need a PostgreSQL database available to the user
```
Comment (coolo) [inline docs/Installing.asciidoc]: I wonder: the previous steps say zypper in openQA from the devel:openQA repo. That zypper command will already give you a working postgresql setup and start the postgresql.service. But adding this here makes the user assume he has to do these steps manually. 

Should we split this into 'from package' and 'from git'?

**Comment by SergioAtSUSE** [comment]
@coolo, @okurz, please review my new changes.

If you don't have anything more against the PR, I will then rebase and make it presentable.

File: docs/Installing.asciidoc [hunk 4]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
...
-The easiest way to install openQA is from packages. You can find openSUSE
-packages in OBS in the
-https://build.opensuse.org/project/show/devel:openQA[openQA:devel] repository.
+Take in mind that there are a lot of disruptive changes between openQA
+versions, you need to be sure that the webui and the worker that you are using
+have the same version number or, at least, are compatible.
 
-For Fedora, packages are available in the official repositories for Fedora 23
-and later. Installation on these distributions is therefore pretty simple:
+For exampe, the package distributed with Leap 42.3 is not compatible with the
```
Comment (okurz) [inline docs/Installing.asciidoc]: you should mention openSUSE here so "openSUSE Leap 42.3"

File: docs/Installing.asciidoc [hunk 5]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
...
+version on Tumbleweed.
+And the package distributed with Tumbleweed may not be compatible with the
+version in development package.
```
Comment (okurz) [inline docs/Installing.asciidoc]: in *the* development package

File: docs/Installing.asciidoc [hunk 6]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
...
+=== Installation from distribution packages
 
-# openSUSE Leap 42.3
-zypper ar -f obs://devel:openQA/openSUSE_Leap_42.3 openQA
-zypper ar -f obs://devel:openQA:Leap:42.3/openSUSE_Leap_42.3 openQA-perl-modules
+The easiest way to install openQA is from distribution packages.
 
-# openSUSE Tumbleweed
-zypper ar -f obs://devel:openQA/openSUSE_Tumbleweed openQA
+- For openSUSE, packages are available for Leap 42.3 and Later.
```
Comment (okurz) [inline docs/Installing.asciidoc]: s/Later/later

File: docs/Installing.asciidoc [hunk 7]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
...
+
 # Fedora 23+
 dnf install openqa openqa-httpd
-----
+-------------------------------------------------------------------------------
+
+
+=== Installation from development packages
+
+You can find the development packages of openQA in OBS in the
```
Comment (okurz) [inline docs/Installing.asciidoc]: "development packages" might be misleading. How about "development versions of packages"?

File: docs/Installing.asciidoc [hunk 8]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
...
+
+To add the development repository to your system, you can use these commands.
+
+[source,sh]
+-------------------------------------------------------------------------------
+# openSUSE Tumbleweed
+zypper ar -f obs://devel:openQA/openSUSE_Tumbleweed devel-openQA
+
+
+# Leap 42.3
```
Comment (okurz) [inline docs/Installing.asciidoc]: when in L63 we write "openSUSE Tumbleweed" then this should be consistent -> "openSUSE Leap 42.3"

File: docs/Installing.asciidoc [hunk 9]
```diff
@@ -99,13 +135,63 @@ You can do that in +/etc/openqa/openqa.ini+:
 httpsonly = 0
 --------------------------------------------------------------------------------
 
+=== Database
+
+Older versions of openQA, like the one distributed on Leap 42.3, had support for
```
Comment (okurz) [inline docs/Installing.asciidoc]: s/Leap 42.3/openSUSE Leap 42.3/

I would rephrase the section to start with the current version describing postgreSQL and then mention older versions as the exception

File: docs/Installing.asciidoc [hunk 10]
```diff
@@ -99,13 +135,63 @@ You can do that in +/etc/openqa/openqa.ini+:
...
+_4.5.1512500474.437cc1c7_ and now the default database is PostgreSQL.
+
+If you are installing an older version than _4.5.1512500474.437cc1c7_ of
+openQA, SQLite is the default database, and is ready to use, but it is
+recommended to move to PostgreSQL.
+
+===== Configure PostgreSQL
+
+This is intended for openQA versions older than _4.5.1512500474.437cc1c7_.
+Since newer versions already provide a PostgreSQL databse ready to use.
```
Comment (okurz) [inline docs/Installing.asciidoc]: s/databse/database/

File: docs/Installing.asciidoc [hunk 11]
```diff
@@ -99,13 +135,66 @@ You can do that in +/etc/openqa/openqa.ini+:
...
+want to use PostgreSQL, you need to create a database (openqa) and a dedicated
+user account (geekotest) with full access to it.
+
+[source,sh]
+-------------------------------------------------------------------------------
+# Assuming that you already have installed PostgreSQL in your system
+su - postgres -c 'createuser -D geekotest'
+su - postgres -c 'createdb -O geekotest openqa'
+-------------------------------------------------------------------------------
+
```
Comment (SergioAtSUSE) [inline docs/Installing.asciidoc]: When https://github.com/os-autoinst/openQA/pull/1530 gets merged, crosslink here "Migrating from existing SQLite instance to PostgreSQL.

File: docs/Installing.asciidoc [hunk 12]
```diff
@@ -99,13 +135,66 @@ You can do that in +/etc/openqa/openqa.ini+:
...
+--------------------------------------------------------------------------------
+[production]
+dsn = dbi:Pg:dbname=openqa;host=db.example.org
+user = openqa
+password = somepassword
+--------------------------------------------------------------------------------
+
+====== Older versions of openQA
+
+Older versions of openQA, like the one distributed on Leap 42.3, had support for
```
Comment (coolo) [inline docs/Installing.asciidoc]: I don't want documentation about the past. The documentation of old versions is in old versions of openQA-doc

File: docs/Installing.asciidoc [hunk 13]
```diff
@@ -18,34 +18,70 @@ assumed that the reader is already familiar with openQA and has already read the
 Starter Guide, available at the
 https://github.com/os-autoinst/openQA[official repository].
 
+
```
Comment (okurz) [inline docs/Installing.asciidoc]: why?

File: docs/Installing.asciidoc [hunk 14]
```diff
@@ -99,13 +135,66 @@ You can do that in +/etc/openqa/openqa.ini+:
 httpsonly = 0
 --------------------------------------------------------------------------------
 
+=== Database
+
+Since verion _4.5.1512500474.437cc1c7_ of openQA, PostgreSQL is used as the
+default database.
+
+It is possible to configure other databases.
```
Comment (okurz) [inline docs/Installing.asciidoc]: Well, it's not … assuming that "other databases" mean some other than postgreSQL

File: docs/Installing.asciidoc [hunk 15]
```diff
@@ -99,13 +135,66 @@ You can do that in +/etc/openqa/openqa.ini+:
 httpsonly = 0
 --------------------------------------------------------------------------------
 
+=== Database
+
+Since verion _4.5.1512500474.437cc1c7_ of openQA, PostgreSQL is used as the
```
Comment (okurz) [inline docs/Installing.asciidoc]: s/verion/version/

File: docs/Installing.asciidoc (hunk 12 above)
Comment (SergioAtSUSE) [reply docs/Installing.asciidoc]: Leap 42.3 is the present of the "really stable" openSUSE, and this package distributed is this oldie.
Would you prefer a section for Leap 42.3 pitfalls?

File: docs/Installing.asciidoc (hunk 13 above)
Comment (SergioAtSUSE) [reply docs/Installing.asciidoc]: No special reason, I did it for me to add readability. I can remove it.

File: docs/Installing.asciidoc (hunk 14 above)
Comment (SergioAtSUSE) [reply docs/Installing.asciidoc]: That's it what I was told... (@foursixnine ?)
Then removing the line.

File: docs/Installing.asciidoc (hunk 12 above)
Comment (coolo) [reply docs/Installing.asciidoc]: not here. This sounds more like the topic of a personal blog.

**Comment by SergioAtSUSE** [comment]
@okurz, @coolo, please review again

File: docs/Installing.asciidoc [hunk 16]
```diff
@@ -20,32 +20,67 @@ https://github.com/os-autoinst/openQA[official repository].
 
 == Installation
 
-The easiest way to install openQA is from packages. You can find openSUSE
-packages in OBS in the
+Take in mind that there are a lot of disruptive changes between openQA
```
Comment (okurz) [inline docs/Installing.asciidoc]: Better "Keep in mind that there can be disruptive changes between openQA versions,"

File: docs/Installing.asciidoc [hunk 17]
```diff
@@ -99,13 +134,48 @@ You can do that in +/etc/openqa/openqa.ini+:
 httpsonly = 0
 --------------------------------------------------------------------------------
 
+=== Database
+
+Since version _4.5.1512500474.437cc1c7_ of openQA, PostgreSQL is used as the
+default database.
```
Comment (coolo) [inline docs/Installing.asciidoc]: remove the 'default' here - because:

**Comment by SergioAtSUSE** [comment]
@coolo, ok, that should be.

Finger crossed
